Fix GCC 10.2.0 build on Ubuntu Hirsute s390x

sergiodj opened the pull request:
GCC reports
 error: free-nonheap-object: snmp_core.cc(950): snmpCreateOidFromStr

yadij commented:
NP: updated a description mistake calling the free'd variable a heap variable when it is on stack.

yadij commented:
OK to test

sergiodj commented:
> NP: updated a description mistake calling the free'd variable a heap variable when it is on stack.

You're correct, sorry about the mistake.  I've fixed the commit message now.

rousskov reviewed (approved):
Thank you for fixing this!

I simplified PR title/description. The title should reflect the high-level effects of the changes. As for the PR description, there is no need to retell in complicated words what the diff already clearly shows to any developer.

It would be useful to document what the build failure (i.e. the compiler error name or text) was, but I need @sergiodj help for that -- I do not know what the error was.

I am temporary removing the `M-cleared-for-merge` label to give @yadij a chance to polish or undo my PR title/description changes and to give @sergiodj a chance to supply that compiler error information.

yadij commented:
The error message is what was used as title before I updated it. Added it to the description, though Anubis already has the merge commit so any changes here may not be picked up.

sergiodj commented:
> It would be useful to document what the build failure (i.e. the compiler error name or text) was, but I need @sergiodj help for that -- I do not know what the error was.

The build failure can be seen here:

https://launchpadlibrarian.net/531216335/buildlog_ubuntu-hirsute-s390x.squid_4.13-1ubuntu2_BUILDING.txt.gz

Here's the excerpt:

```
libtool: link: s390x-linux-gnu-g++ -Wall -Wpointer-arith -Wwrite-strings -Wcomments -Wshadow -Woverloaded-virtual -Werror -pipe -D_REENTRANT -I/usr/include/libxml2 -I/usr/include/p11-kit-1 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security .libs/squidS.o -Wl,-Bsymbolic-functions -flto=auto -Wl,-z -Wl,relro -Wl,-z -Wl,now -o squid AclRegs.o AuthReg.o AccessLogEntry.o AsyncEngine.o cache_cf.o CacheDigest.o cache_manager.o CachePeer.o carp.o cbdata.o client_db.o client_side.o client_side_reply.o client_side_request.o BodyPipe.o clientStream.o CollapsedForwarding.o CommandLine.o CompletionDispatcher.o ConfigOption.o ConfigParser.o CpuAffinity.o CpuAffinityMap.o CpuAffinitySet.o debug.o delay_pools.o DelayId.o DelayBucket.o DelayConfig.o DelayPool.o DelaySpec.o DelayTagged.o DelayUser.o DelayVector.o ClientDelayConfig.o fs_io.o dlink.o dns_internal.o Downloader.o errorpage.o ETag.o event.o EventLoop.o external_acl.o ExternalACLEntry.o FadingCounter.o fatal.o fd.o fde.o filemap.o fqdncache.o FwdState.o gopher.o helper.o htcp.o http.o HttpHdrCc.o HttpHdrRange.o HttpHdrSc.o HttpHdrScTarget.o HttpHdrContRange.o HttpHeader.o HttpHeaderTools.o HttpBody.o HttpControlMsg.o HttpMsg.o HttpReply.o RequestFlags.o HttpRequest.o icp_v2.o icp_v3.o Instance.o int.o internal.o ipc.o ipcache.o LogTags.o main.o MasterXaction.o mem_node.o MemBuf.o MemObject.o mime.o mime_header.o multicast.o neighbors.o Notes.o Parsing.o pconn.o peer_digest.o peer_proxy_negotiate_auth.o peer_select.o peer_sourcehash.o peer_userhash.o PeerPoolMgr.o Pipeline.o redirect.o refresh.o RemovalPolicy.o send-announce.o SBufStatsAction.o snmp_core.o snmp_agent.o SquidMath.o SquidNew.o stat.o StatCounters.o StatHist.o String.o StrList.o stmem.o store.o StoreFileSystem.o store_io.o StoreIOState.o store_client.o store_digest.o store_key_md5.o store_log.o store_rebuild.o store_swapin.o store_swapmeta.o store_swapout.o StoreMetaUnpacker.o StoreMeta.o StoreMetaMD5.o StoreMetaSTD.o StoreMetaSTDLFS.o StoreMetaURL.o StoreMetaVary.o StoreStats.o StoreSwapLogData.o Transients.o MemStore.o time.o tools.o tunnel.o unlinkd.o urn.o wccp.o wccp2.o whois.o wordlist.o XactionInitiator.o LoadableModule.o LoadableModules.o err_type.o err_detail_type.o globals.o hier_code.o icp_opcode.o lookup_t.o repl_modules.o swap_log_op.o -Wl,--export-dynamic  auth/.libs/libacls.a ident/.libs/libident.a acl/.libs/libacls.a acl/.libs/libstate.a auth/.libs/libauth.a acl/.libs/libapi.a clients/.libs/libclients.a servers/.libs/libservers.a ftp/.libs/libftp.a helper/.libs/libhelper.a http/.libs/libhttp.a dns/.libs/libdns.a base/.libs/libbase.a ./.libs/libsquid.a ip/.libs/libip.a fs/.libs/libfs.a DiskIO/.libs/libdiskio.a -lpthread comm/.libs/libcomm.a anyp/.libs/libanyp.a security/.libs/libsecurity.a ipc/.libs/libipc.a mgr/.libs/libmgr.a parser/.libs/libparser.a eui/.libs/libeui.a icmp/.libs/libicmp.a log/.libs/liblog.a format/.libs/libformat.a sbuf/.libs/libsbuf.a repl/liblru.a repl/libheap.a -lnettle -lcrypt adaptation/.libs/libadaptation.a -lecap esi/.libs/libesi.a ../lib/libTrie/libTrie.a -lxml2 -lexpat snmp/.libs/libsnmp.a ../lib/snmplib/.libs/libsnmplib.a mem/.libs/libmem.a store/.libs/libstore.a ../lib/.libs/libmisccontainers.a ../lib/.libs/libmiscencoding.a ../lib/.libs/libmiscutil.a -lgnutls -L/usr/lib/s390x-linux-gnu/mit-krb5 -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err ../compat/.libs/libcompatsquid.a -lsystemd -lm -lnsl -lresolv -lcap -lnetfilter_conntrack -lrt -L.. -lltdl
In function 'free_const',
    inlined from 'snmpCreateOidFromStr' at snmp_core.cc:950:5,
    inlined from 'snmpLookupNodeStr' at snmp_core.cc:899:31:
../compat/xalloc.cc:175:9: error: attempt to free a non-heap object 'name' [-Werror=free-nonheap-object]
  175 |     free(s);
      |         ^
In function 'free_const',
    inlined from 'snmpCreateOidFromStr' at snmp_core.cc:950:5,
    inlined from 'snmpAddNodeStr' at snmp_core.cc:974:31:
../compat/xalloc.cc:175:9: error: attempt to free a non-heap object 'n' [-Werror=free-nonheap-object]
  175 |     free(s);
      |         ^
lto1: all warnings being treated as errors
make[5]: *** [/tmp/cc6LDcoQ.mk:95: /tmp/squid.An03ub.ltrans31.ltrans.o] Error 1
make[5]: *** Waiting for unfinished jobs....
lto-wrapper: fatal error: make returned 2 exit status
compilation terminated.
/usr/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
```

Thank you for the review and the adjustments to the commit message.

sergiodj commented:
I've updated the commit message with the error.  Let me know if that's better.
